### PR TITLE
Update readme to define GraphStorm.

### PR DESCRIPTION
**Pull request by zheng-da**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,15 @@
 ## GraphStorm
-The GraphStorm is a graph machine learning (GML) framework for enterprise use cases. It simplifies the development, training and deployment of GML models for industry-scale graphs. It provides scalable training and inference pipelines for various GML models and is optimized for extremely large graphs (measured in billons of nodes and edges). To help develop SOTA models, the framework allows very large degree of customization by providing a large collection of configurations for customizing model implementations and training pipelines as well as modularization in the training pipeline that allows users to replace every component in the pipeline.
+The GraphStorm is a graph machine learning (GML) framework for enterprise use cases.
```
Comment (thvasilo) [inline README.md:2]:
```suggestion
GraphStorm is a graph machine learning (GML) framework for enterprise use cases.
```

File: README.md [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
+It simplifies the development, training and deployment of GML models for industry-scale graphs.
+It provides scalable graph loading, training and inference pipelines for various GML models
+and is optimized for extremely large graphs (measured in billons of nodes and edges).
+GraphStorm provides a collection of built-in GML models and users can train a GML model
+with a single command line without writing any code. To help develop SOTA models,
+GraphStorm provides a large collection of arguments to configure model implementations
```
Comment (thvasilo) [inline README.md:8]: arguments or modules here?

File: README.md [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+and training pipelines to improve model performance. GraphStorm also provides a programming
+interface to train any custom GML models in a distributed manner. In this case, users
```
Comment (thvasilo) [inline README.md:10]:
```suggestion
interface to train any custom GML model in a distributed manner. In this case, users
```

**Review by thvasilo — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (zheng-da) [reply README.md:8]: I think it's arguments. what do you mean by module?

Comment (thvasilo) [reply README.md]: Usually we relate "arguments" to a list of arguments (values) to be passed to a script or function. Are we saying that similar to say the `nn` module of Pytorch allows developers to easily create NNs, GSF provides the building blocks for GNNs?
Or are we talking about actual arguments passed to a script/function?

File: README.md [hunk 4]
```diff
@@ -1,5 +1,14 @@
 ## GraphStorm
-The GraphStorm is a graph machine learning (GML) framework for enterprise use cases. It simplifies the development, training and deployment of GML models for industry-scale graphs. It provides scalable training and inference pipelines for various GML models and is optimized for extremely large graphs (measured in billons of nodes and edges). To help develop SOTA models, the framework allows very large degree of customization by providing a large collection of configurations for customizing model implementations and training pipelines as well as modularization in the training pipeline that allows users to replace every component in the pipeline.
+GraphStorm is a graph machine learning (GML) framework for enterprise use cases.
+It simplifies the development, training and deployment of GML models for industry-scale graphs.
+It provides scalable training and inference pipelines for various Graph Machine Learning (GML) models
+and is optimized for extremely large graphs (measured in billons of nodes and edges).
```
Comment (thvasilo) [inline README.md:5]: I think these two sentences are saying the same thing?

I think you can remove line 3 and keep the more descriptive sentence in lines 4-5.

File: README.md [hunk 5]
```diff
@@ -1,5 +1,14 @@
...
+GraphStorm provides a collection of built-in GML models and users can train a GML model
```
Comment (thvasilo) [inline README.md:6]: Can add a line break above here to break up into two paragraphs?

**Review by thvasilo — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (zheng-da) [reply README.md]: i'm talking about the actual arguments passed to a script.

File: README.md (hunk 4 above)
Comment (zheng-da) [reply README.md:5]: they are different. line 3 talks about the simplicity of developing, training and inference GML models in the context of industry graphs.
line 4 and 5 emphasize the scalability.

File: README.md (hunk 5 above)
Comment (zheng-da) [reply README.md:6]: the topic sentence of the paragraph is "GraphStorm is a graph machine learning (GML) framework for enterprise use cases."
this paragraph isn't very long. breaking it into two paragraphs will make each paragraph very short.

**Pull request merged**